pkg/kotsadm/objects/minio_objects.go

```diff
@@ -217,6 +213,255 @@ func MinioStatefulset(deployOptions types.DeployOptions, size resource.Quantity)
 	return statefulset, nil
 }
 
+func migrateMinioXlInitContainers(deployOptions types.DeployOptions, resourceRequirements corev1.ResourceRequirements) []corev1.Container {
+	volumeMounts := append(minioVolumeMounts(), minioXlMigrationVolumeMounts()...)
+
+	return []corev1.Container{
+		{
+			Image:           GetAdminConsoleImage(deployOptions, "minio-client"),
+			ImagePullPolicy: corev1.PullIfNotPresent,
+			Name:            "copy-minio-client",
+			Command: []string{
+				"/scripts/copy-minio-client.sh",
+			},
+			VolumeMounts: volumeMounts,
+			Env: []corev1.EnvVar{
+				{
+					Name:  "KOTSADM_MINIO_MIGRATION_WORK_DIR",
+					Value: "/migration-work",
+				},
+			},
+			Resources:       resourceRequirements,
+			SecurityContext: secureContainerContext(deployOptions.StrictSecurityContext),
+		},
+		{
+			Image:           deployOptions.MigrateMinioXlOldImage,
+			ImagePullPolicy: corev1.PullIfNotPresent,
+			Name:            "export-minio-data",
+			Command: []string{
+				"/scripts/export-minio-data.sh",
+			},
+			VolumeMounts: volumeMounts,
+			Env: []corev1.EnvVar{
+				{
+					Name:  "KOTSADM_MINIO_ENDPOINT",
+					Value: "http://localhost:9000",
+				},
+				{
+					Name:  "KOTSADM_MINIO_BUCKET_NAME",
+					Value: "kotsadm",
+				},
+				{
+					Name: "MINIO_ACCESS_KEY",
+					ValueFrom: &corev1.EnvVarSource{
+						SecretKeyRef: &corev1.SecretKeySelector{
+							LocalObjectReference: corev1.LocalObjectReference{
+								Name: "kotsadm-minio",
+							},
+							Key: "accesskey",
+						},
+					},
+				},
+				{
+					Name: "MINIO_SECRET_KEY",
+					ValueFrom: &corev1.EnvVarSource{
+						SecretKeyRef: &corev1.SecretKeySelector{
+							LocalObjectReference: corev1.LocalObjectReference{
+								Name: "kotsadm-minio",
+							},
+							Key: "secretkey",
+						},
+					},
+				},
+				{
+					Name:  "KOTSADM_MINIO_LEGACY_ALIAS",
+					Value: "legacy",
+				},
+				{
+					Name:  "KOTSADM_MINIO_MIGRATION_WORK_DIR",
+					Value: "/migration-work",
```

Comment:
nit pick: i would've called this `/migration`

Reply:
same as above :)